tools/fixedpointarithmetic/src/test/java/dk/alexandra/fresco/fixedpoint/basic/BasicArithmeticTests.java

```diff
@@ -0,0 +1,339 @@
+
+    @Override
+    public TestThread<ResourcePoolT, ProtocolBuilderNumeric> next() {
+      BigDecimal value = BigDecimal.valueOf(10.00100);
+      BigDecimal value2 = BigDecimal.valueOf(0.2);
+      return new TestThread<ResourcePoolT, ProtocolBuilderNumeric>() {
+        @Override
+        public void test() throws Exception {
+          Application<BigDecimal, ProtocolBuilderNumeric> app = producer -> {
+            FixedNumeric numeric = new DefaultFixedNumeric(producer, 5);
+
+            DRes<SFixed> input2 = numeric.input(value2, 1);
+            DRes<SFixed> product = numeric.mult(value, input2);
+
+            return numeric.open(product);
+          };
+          BigDecimal output = runApplication(app);
+          Assert.assertEquals(value.multiply(value2)
+              .setScale(5, RoundingMode.HALF_UP), output);
+        }
+      };
+    }
+  }
+  
+  public static class TestDivisionSecret<ResourcePoolT extends ResourcePool>
+      extends TestThreadFactory<ResourcePoolT, ProtocolBuilderNumeric> {
+
+    @Override
+    public TestThread<ResourcePoolT, ProtocolBuilderNumeric> next() {
+      BigDecimal value = BigDecimal.valueOf(10.00100);
+      BigDecimal value2 = BigDecimal.valueOf(0.2);
+      return new TestThread<ResourcePoolT, ProtocolBuilderNumeric>() {
+        @Override
+        public void test() throws Exception {
+          Application<BigDecimal, ProtocolBuilderNumeric> app = producer -> {
+            FixedNumeric numeric = new DefaultFixedNumeric(producer, 5);
+
+            DRes<SFixed> input = numeric.input(value, 1);
+            DRes<SFixed> input2 = numeric.input(value2, 1);
+            DRes<SFixed> product = numeric.div(input, input2);
+
+            return numeric.open(product);
+          };
+          BigDecimal output = runApplication(app);
+          Assert.assertEquals(value.divide(value2)
+              .setScale(5, RoundingMode.HALF_UP), output);
+        }
+      };
+    }
+  }
+  
+  public static class TestDivisionKnownDivisor<ResourcePoolT extends ResourcePool>
+      extends TestThreadFactory<ResourcePoolT, ProtocolBuilderNumeric> {
+
+    @Override
+    public TestThread<ResourcePoolT, ProtocolBuilderNumeric> next() {
+      BigDecimal value = BigDecimal.valueOf(10.00100);
+      BigDecimal value2 = BigDecimal.valueOf(0.2);
+      return new TestThread<ResourcePoolT, ProtocolBuilderNumeric>() {
+        @Override
+        public void test() throws Exception {
+          Application<BigDecimal, ProtocolBuilderNumeric> app = producer -> {
+            FixedNumeric numeric = new DefaultFixedNumeric(producer, 5);
+
+            DRes<SFixed> input = numeric.input(value, 1);
+            DRes<SFixed> product = numeric.div(input, value2);
+
+            return numeric.open(product);
+          };
+          BigDecimal output = runApplication(app);
+          Assert.assertEquals(value.divide(value2)
+              .setScale(5, RoundingMode.HALF_UP), output);
+        }
+      };
+    }
+  }  
```

Comment:
Maybe rewrite a few of the tests to be a list of product/sums that are tested instead of just two single values